Mathlib/Data/Fintype/Sigma.lean

```diff
@@ -43,3 +43,11 @@ instance PSigma.instFintype : Fintype (Σ' i, κ i) := .ofEquiv _ (Equiv.psigmaE
 
 @[simp] lemma Finset.univ_sigma_univ : univ.sigma (fun _ ↦ univ) = (univ : Finset (Σ i, κ i)) := rfl
 #align finset.univ_sigma_univ Finset.univ_sigma_univ
+
+instance Infinite.sigma_of_left {β : α → Type*} [Infinite α] [∀ a, Nonempty (β a)] :
+    Infinite ((a : α) × (β a)) :=
+  Infinite.of_surjective Sigma.fst Sigma.fst_surjective
+
+theorem Infinite.sigma_of_right {β : α → Type*} {a : α} [Infinite (β a)] :
```

Comment:
You can also keep a version of your non-instance, but the following can be made an instance
```suggestion
instance Infinite.sigma_of_right {β : α → Type*} [Nonempty α] [∀ a, Infinite (β a)] :
```